### ISSUE TITLE
学習・自己対局の開発履歴を可視化する Streamlit ダッシュボードを作る

### ISSUE THREAD

**Issue by LoveKapibarasan** [LoveKapibarasan/python-dlshogi2#1]
## 目的

学習・自己対局の「開発履歴」を後から振り返れるように、Streamlit などで可視化ダッシュボードを作る。

現状、学習の進捗は `pydlshogi2/train.py` の `logging` 出力（`--log` で指定するテキストログ）にしか残っていない。

```
epoch = 1, steps = 1000, train loss = ..., ..., ..., test loss = ..., ..., ..., test accuracy = ..., ...
```

そのため以下ができない。

- 複数 run（SL の学習、`rl_loop.sh` の各イテレーション）を横並びで比較する
- loss / accuracy の推移をグラフで見る
- どのハイパーパラメータ（`--lr` / `--batchsize` / `--blocks` / `--channels` / `--val_lambda` など）の run だったかを後から特定する

### 1. メトリクスを構造化して保存する

テキストログのままだとパースが面倒なので、機械可読な形式で残す。

- [ ] `train.py` に `--metrics <path.jsonl>` のようなオプションを追加し、1 行 1 レコードの JSONL を追記する
  - `{"run_id", "timestamp", "epoch", "step", "train_loss_policy", "train_loss_value", "train_loss_total", "test_loss_*", "test_accuracy_policy", "test_accuracy_value"}`
- [ ] run 開始時に 1 度だけ run メタ情報（argparse の全引数、git commit hash、ホスト名、GPU 名）を書き出す
- [ ] 既存のテキストログ出力はそのまま残す（後方互換）
- [ ] 代替案として TensorBoard (`SummaryWriter`) も検討する。ただし run 間比較と自己対局の統計まで扱いたいので、JSONL + Streamlit を第一候補とする

### 2. 自己対局側のメトリクスも記録する

`rl_loop.sh` / `pydlshogi2/selfplay.py` から、イテレーションごとに以下を残す。

- [ ] 生成局面数・対局数、平均手数、先手/後手/引き分けの勝率分布
- [ ] 1 対局あたりの所要時間、playout 数などの設定値
- [ ] イテレーション番号 → 使用チェックポイント → 生成 hcpe → 学習後チェックポイント の対応表

### 3. Streamlit ダッシュボード

- [ ] `dashboard/app.py`（仮）を追加。`streamlit run dashboard/app.py` で起動
- [ ] 画面構成の案
  - **Runs 一覧**: run_id、開始時刻、commit、主要ハイパラ、最終 test accuracy をテーブル表示
  - **学習曲線**: 選択した複数 run の train/test loss、policy/value accuracy を step 軸で重ね描き
  - **RL ループ**: イテレーションごとの自己対局統計と、学習後の指標の推移
  - **チェックポイント**: `checkpoints/` 配下のファイル一覧（サイズ、更新時刻、埋め込まれた architecture）
- [ ] データソースはローカルの JSONL ディレクトリを指定する方式（DB は当面不要）
- [ ] `requirements.txt` とは分離した `dashboard/requirements.txt` に streamlit / pandas / altair を置き、エンジン本体の依存を増やさない

### 4. ドキュメント

- [ ] README に「Dashboard」節を追加（起動方法・スクリーンショット）
- [ ] `docs/guides/` に使い方ガイドを追加

## 補足・検討事項

- Vast.ai などのリモートで学習する運用があるため、メトリクス JSONL はチェックポイントと同じディレクトリに置き、まとめて回収できるようにしたい
- preemption からの `--resume` 時に run_id を引き継ぐか、新しい run として記録して親 run へのリンクを持たせるかを決める必要がある
